conbench/entities/_comparator.py

```diff
@@ -175,71 +172,54 @@ def tags(self):
-            "baseline_z_score": to_float(self.baseline_z_score),
-            "contender_z_score": to_float(self.contender_z_score),
-            "baseline_z_regression": self.baseline_z_regression,
-            "baseline_z_improvement": self.baseline_z_improvement,
-            "contender_z_regression": self.contender_z_regression,
-            "contender_z_improvement": self.contender_z_improvement,
-            "baseline": fmt(baseline),
-            "contender": fmt(contender),
-            "baseline_error": self.baseline.error if self.baseline else None,
-            "contender_error": self.contender.error if self.contender else None,
-            "baseline_id": self.baseline.id if self.baseline else None,
-            "contender_id": self.contender.id if self.contender else None,
-            "baseline_batch_id": self.baseline.batch_id if self.baseline else None,
-            "contender_batch_id": self.contender.batch_id if self.contender else None,
-            "baseline_run_id": self.baseline.run_id if self.baseline else None,
-            "contender_run_id": self.contender.run_id if self.contender else None,
             "unit": self.unit,
             "less_is_better": self.less_is_better,
-            "tags": self.tags,
+            "baseline": {
+                "benchmark_name": self.baseline.benchmark_name,
+                "case_permutation": self.baseline.case_permutation,
+                "language": self.baseline.language,
+                "value": _round(to_float(baseline)),
+                "error": self.baseline.error,
+                "benchmark_result_id": self.baseline.id,
+                "batch_id": self.baseline.batch_id,
+                "run_id": self.baseline.run_id,
+                "tags": self.baseline.tags,
+            }
+            if self.baseline
+            else None,
+            "contender": {
+                "benchmark_name": self.contender.benchmark_name,
+                "case_permutation": self.contender.case_permutation,
+                "language": self.contender.language,
+                "value": _round(to_float(contender)),
+                "error": self.contender.error,
+                "benchmark_result_id": self.contender.id,
+                "batch_id": self.contender.batch_id,
+                "run_id": self.contender.run_id,
+                "tags": self.contender.tags,
+            }
+            if self.contender
+            else None,
+            "analysis": {
+                "pairwise": {
+                    "percent_change": _round(to_float(self.change * 100)),
+                    "percent_threshold": to_float(self.threshold),
+                    "regression_indicated": self.regression,
+                    "improvement_indicated": self.improvement,
+                },
+                "lookback_z_score": {
+                    "z_threshold": to_float(self.threshold_z),
+                    "z_score": _round(to_float(self.contender.z_score))
```

Comment:
I think there may be places outside this file that make `z_score` 0.0 if it actually should be missing. I don't think we should ever return 0.0 unless it's actually 0.0 (i.e. all ≥2 historic observations have the same value for some reason). Would you mind going through and ensuring that doesn't happen, and maybe adding a test? (It could be a separate PR)

Reply:
Made #1128 because that change would have to happen outside the scope of `_comparator.py`, which otherwise is where all the behavioral changes of this PR are